### ISSUE TITLE
Console

### ISSUE THREAD

**Issue by alessaba** [SquaredTiki/Dringend-Issues#70]
Is it possible to make a console with warnings,errors and NSLogs?
it would be so useul :-)

**Comment by jmurff** [comment]
This would be cool but I don't see how he can make that happen. The app is installed once that happens your  sandboxed. There are someways to access the devices console programatically.

I did some research googling "ios access device console programmatically". There is some stuff you can do within your own app... I thought this one was particularly  interesting. https://github.com/nicklockwood/iConsole"

**Comment by alessaba** [comment]
I don't know if it can be useful but there's an app that reads logs
https://itunes.apple.com/it/app/consolekit/id648875678?mt=8
it is not the only one....

Filippo Claudi

**Comment by SquaredTiki** [comment]
@filippocld Whilst it was possible in the past to access the device log, as of iOS 7 it is no longer. That app you linked does not work on iOS 7. There are alternatives I am considering for collecting the NSLog outputs from installed apps however which may be viable.

**Comment by alessaba** [comment]
uh :-(
ok..i'll wait :-)

Filippo Claudi

**Comment by jmurff** [comment]
iConsole seems to work for me to bring up an in app console.

**Comment by alessaba** [comment]
Yes.. but the goal is printing log of another app in Dringend...

Filippo Claudi

**Comment by ezhes** [comment]
I've implemented this into my own app and it should be really simple to build this into Dringend. The setup utilizes a server (which would be implmented into the Dringend app) and a client framework. The client framework sends logs to the server over HTTP and then the server appliction can trigger background banner notifications. You can also add something like log levels, filtering, and other fancy features that are not normally offfered by a standard text console.

**Comment by SquaredTiki** [comment]
@shusain93 Thanks for the details and indeed this is the most feasible way at the moment. The issue will be automatically integrating such a framework into the users projects.

**Comment by ezhes** [comment]
It doesn't need to be automatic, I suppose. I haven't done much when it comes to manually adding frameworks to the Xcode configure, but wouldn't it be possible to just add a framework like normal, open the file as text, find all the places Xcode put the framework, and then whenever someone wants to add the framework automatically the app can just insert the framework decorations into the spots you found before. I think you would end up needing to hardcore some value to look for, which would cause issues with SDK changes
